src/map/Map.cpp

```diff
@@ -0,0 +1,239 @@
+#include "Map.h"
+
+using namespace std;
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: Map(string file)
+ * Parameters:
+ *      file - File name, must be a CSV File
+ * Description:
+ * Map Object Constructor, creates a map object using the map selected file name.
+ */
+Map::Map(string file):fname(file) {
+    for(int i = 0; i < M_HEIGHT; i++) {
+        for(int j = 0; j < M_WIDTH; j++) {
+            AIMap[i][j] = 0;
+            mapdata[i][j] = 0;
+        }
+    }
+}
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: int loadFileData()
+ * Description:
+ * Read's the Map CSV file map data with the appropriate tile IDs
+ */
+int Map::loadFileData() {
+    // Open file stream for selected map file
+    ifstream file;
+    file.open(fname.c_str());
+
+    if (file.fail()) {
+        printf("failed to open file\n");
+        return 0;
+    }
+
+    // Array of wallstart positions
+    struct MapPoint wallStart[MAX_WALLS];
+    // Indices
+    int j = 0;
+    int i = 0;
+
+    // Character Variable
+    char ch;
+    while (file >> noskipws >> ch) {
+        switch (ch) {
+            case ',':
+                continue;
+            case '\n':
+                j = 0;
+                i++;
+                continue;
+            case WALL:              //Wall piece found, update AI Map
+                AIMap[i][j] = 1;
+                break;
+            case WALL_START:        //Label start of wall rectangle
+                wallStart[ws].x = j;
+                wallStart[ws].y = i;
+                ws++;
+                AIMap[i][j] = 1;
+                break;
+            // case CONCRETE_START:    //Start of concrete area
+            case ZOMBIE_SPAWN:      //Zombie spawn Points
+                zombieSpawn[zs].x = j;
+                zombieSpawn[zs].y = i;
+                zs++;
+                break;
+            case SHOP_SPOT:         // Shop spot points
+                shops[s].x = j;
+                shops[s].y = i;
+                s++;
+                break;
+            case BASE_START:        //Start of base area
+                base.x = j;
+                base.y = i;
+                break;
+        }
+        mapdata[i][j] = ch;               //Store map file value
+        j++;
+    }
+    file.close();
+
+    genWalls(wallStart);
+
+    // printData();
+    return 1;
+}
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: void genWalls(const struct MapPoint wstart[MAX_WALLS])
+ * Parameters:
+ *      wstart - Array of wall start MapPoints.
+ * Description:
+ * Read's the Map CSV file map data with the appropriate tile IDs
+ */
+void Map::genWalls(const struct MapPoint wstart[MAX_WALLS]) {
+    for (int q = 0; q < ws; q++) {
+        int startx = wstart[q].x;
+        int starty = wstart[q].y;
+        int nextx = startx + 1;
+        int nexty = starty + 1;
+        int endx;
+        int endy;
```

Comment:
Only declare variables near where they are used
If they are not changed, they should be const
And if you assign them only once in a loop, declare them non-const outside the loop so you're not destroying ints every loop iteration